Doc/library/argparse.rst

```diff
@@ -2162,6 +2162,46 @@ Intermixed parsing
 
    .. versionadded:: 3.7
 
+Registering custom types or actions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. method:: ArgumentParser.register(registry_name, value, object)
+
+   It is possible to register custom actions, types, or other objects with
```

Comment:
Looking at this text alone, without the context of the rest of the docs, it sounds really vague to me. What is the registration for? Improved error messages? Anything else? The example definitely helps, but it would be nice to state this upfront.